src/main/java/org/jabref/logic/cleanup/RenamePdfCleanup.java

```diff
@@ -138,7 +138,15 @@ public RenamePdfCleanup(boolean onlyRelativePaths, BibDatabaseContext databaseCo
                 // We use the file directory (if none is set - then bib file) to create relative file links.
                 // The .get() is legal without check because the method will always return a value.
                 Path settingsDir = databaseContext.getFirstExistingFileDir(fileDirectoryPreferences).get();
-                newFileList.add(new LinkedFile(description, settingsDir.relativize(newPath).toString(), type));
+                try {
+                    newFileList.add(new LinkedFile(description, settingsDir.relativize(newPath).toString(), type));
+                } catch (IllegalArgumentException e) {
+                    if (e.getMessage().equals("'other' has different root")) {
```

Comment:
Although this _might_ work, I don't like the way you mix error handling and control flow. Would it be possible to perform a check **before** any exception is thrown and handle the two cases separately?

Reply:
Yes, it was a bad way to do it. In the new commit, I checked if database directory and pdf directory has different roots without throwing an exception.